### PR TITLE
ddl: Advance iterator when skip handling TiFlash tables

### LINKED ISSUE #45859: Pd's /status/region may be frequently called when TiFlash is not up
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Let TiFlash down after a table is ready.

### 2. What did you expect to see? (Required)

Pd's log print print in about 2 secs per table.

### 3. What did you see instead (Required)

Many logs printed on Pd for single table.

### 4. What is your TiDB version? (Required)

nightly

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #45859

Problem Summary:

In `pollAvailableTableProgress`, iterator of `UpdatingProgressTables` may not be advanced under several errors.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_tiflash_api.go [hunk 1]
```diff
@@ -406,6 +406,8 @@ func pollAvailableTableProgress(schemas infoschema.InfoSchema, _ sessionctx.Cont
 				// https://github.com/pingcap/tidb/issues/39949
 				panic(err)
 			}
+			pollTiFlashContext.UpdatingProgressTables.Remove(element)
```
Comment (zimulala) [inline ddl/ddl_tiflash_api.go:409]: Why does removing elements solve this problem?

Comment (CalvinNeo) [reply ddl/ddl_tiflash_api.go:409]: The outside loop will iterate for `min(pollMaxCount, UpdatingProgressTables.Len())`. If we `continue` here, the loop will carry on with `pollMaxCount -= 1`. However the iterator of `UpdatingProgressTables` will not be advanced. That means the same element will be handled for `pollMaxCount` times.

We have observed many requests of the same table to PD in less than 1 sec.

**Comment by CalvinNeo** [comment]
Need to be cherry-picked to 6.5 and 7.1

File: ddl/ddl_tiflash_api.go (hunk 1 above)
Comment (zimulala) [reply ddl/ddl_tiflash_api.go:409]: Do we have test for this logic? If it hard to add UT, please add a manual test description.
Rest LGTM.

File: ddl/tests/tiflash/ddl_tiflash_test.go [hunk 2]
```diff
@@ -1088,6 +1089,58 @@ func TestTiFlashGroupIndexWhenStartup(t *testing.T) {
...
+	})
+	require.NoError(t, err)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ddl.PollAvailableTableProgress(s.dom.InfoSchema(), sctx, pollTiflashContext)
+	}()
+	time.Sleep(ddl.PollTiFlashInterval)
+	s.tiflash.SetNetworkError(false)
```
Comment (zimulala) [inline ddl/tests/tiflash/ddl_tiflash_test.go:1129]: Could we remove this logic? After this PR, we wouldn't have a dead circle.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**